plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -0,0 +1,657 @@
+<?php
+// phpcs:disable Generic.Commenting.Todo.TaskFound
+/**
+ * OrdersTableQuery class file.
+ */
+
+namespace Automattic\WooCommerce\Internal\DataStores\Orders;
+
+use Automattic\WooCommerce\Internal\Utilities\DatabaseUtil;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * This class provides a `WP_Query`-like interface to custom order tables.
+ */
+class OrdersTableQuery {
```

Comment:
maybe we can refactor this to have a base class for querying custom tables. but perhaps in a separate PR since this is an internal class.

Reply:
Agreed. I didn't want to over-complicate an already complex piece, but that's definitely something we can do for other future custom tables implementations. Sounds like a good idea.